test/state/precompiles.cpp

```diff
@@ -142,6 +143,37 @@ PrecompileAnalysis expmod_analyze(bytes_view input, evmc_revision rev) noexcept
         static_cast<size_t>(mod_len)};
 }
 
+ExecutionResult ecrecover_execute(const uint8_t* input, size_t input_size, uint8_t* output,
+    [[maybe_unused]] size_t output_size) noexcept
+{
+    assert(output_size == 32);
```

Comment:
I think my comment got lost somewhere, re-adding:

@chfast this assertion fails, because the caller sends `4096` as `output_size`. I'm new to this so I'm not sure which part gets it wrong.

Shall we do `assert(output_size >= 32)` and `output_size = 32` instead?

Reply:
Yes, you can change it to `>= 32`. We changed logic a bit recently.

Reply:
pls double check if this makes sense, I adhered to `identity_execute` [a82f0cb](https://github.com/ethereum/evmone/pull/688/commits/a82f0cbf90146a4bf608eecb577b3cfc0ac91a09)